rep-0149.rst

```diff
@@ -505,10 +494,60 @@ Attributes
   
     "You must give any other recipients of the Work or Derivative Works a copy of this License"
 
+ ``source-files="FILENAME-PATTERN"`` *(optional)*
+
+  A filename pattern using the simplified shell glob syntax specified in
+  Section 6.9 of the `Machine-readable debian/copyright file specification V1.0
+  <https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/>`_ and
+  relative to the ``package.xml`` file.
+
+  The filename pattern specifies the source files this license information
+  refers to. The value ``source-files="*"`` refers to all source files of the
+  package, including source files that are downloaded automatically during the
+  build process - for example in the case of so-called *vendor packages*. If
+  the attribute is not specified, the tag again refers to all source files of
+  the package, including downloaded source files.
+
+  While the Machine-readable debian/copyright file specification V1.0 allows
+  multiple filename patterns separated by spaces, only one pattern is allowed
+  here. Use multiple license tags if necessary (but first consider
+  restructuring the folder structure of the source files).
+
+  If the filename patterns of multiple license tags match a particular file,
+  the last tag applies to it - following the logic described in Section 6.9
+  of the `Machine-readable debian/copyright file specification V1.0
+  <https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/>`_.
+  Consequently, more general tags should be given first.
+
+  Finally note that space characters in the source file paths/names - which
+  should be a very rare case - have to be replaced by a ``?`` in the filename
+  pattern as described at the end of Section 6.9 of the `Machine-readable
+  debian/copyright file specification V1.0
+  <https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/>`_.
```

Comment:
That's how the Machine-readable debian/copyright file specification V1.0 states it. We could do differently, but then we would make the process how the package.xml is translated to the copyright file less transparent. And I consider spaces in source file/folder names an absolute exception.
@ct2034 has built a linter for the new package.xml format, which will of course include a check for space characters.

Reply:
If we would simply replace space characters by `?` during the generation of the copyright file, this could even change the set of matched files. This must be prevented. Therefore I would force those developers, who want to use space characters in the filenames/paths, to think about it themselves.